pdns/statbag.cc

```diff
@@ -237,8 +244,8 @@ vector<pair<T, unsigned int> >StatRing<T,Comp>::get() const
 
 void StatBag::registerRingStats(const string& name)
 {
-  declare("ring-" + name + "-size", "Number of entries in the " + name + " ring", [this,name](const std::string&) { return static_cast<uint64_t>(getRingEntriesCount(name)); });
-  declare("ring-" + name + "-capacity", "Maximum number of entries in the " + name + " ring", [this,name](const std::string&) { return static_cast<uint64_t>(getRingSize(name)); });
+  declare("ring-" + name + "-size", "Number of entries in the " + name + " ring", [this,name](const std::string&) { return static_cast<uint64_t>(getRingEntriesCount(name)); }, StatType::counter);
```

Comment:
gauge, I think?

Reply:
@rgacogne raised the same issue here: https://github.com/PowerDNS/pdns/pull/9182#discussion_r433730340

I am happy either way, but following the [prometheus docs](https://prometheus.io/docs/concepts/metric_types/#counter) the ring-buffer `size` metric can be considered a counter. It either increments and saturates or the buffer is `reset()` and `size` is 0.

Since the ring-buffer can be resized, we considered `capacity` as a gauge. But `resize()` will also affect `size` if the resized buffer is smaller than the number of items it currently holds. Which would make `size` a gauge.

Reply:
When considering the case of `resize()` to a `capacity` smaller than the current `size` it is clear, that `size` must be a gauge too. Fixed in the latest commit.

Reply:
ah, sorry, I did see that comment (but did not read it carefully). I see now how the definitions of 'gauge' and 'metric' in Prometheus do not exactly match my intuition of the terms.

resize to smaller but non-zero seems like a sensible argument, thanks!